crates/gateway-types/src/reply.rs

```diff
@@ -1147,14 +1159,31 @@ pub(crate) mod transaction {
                     constructor_calldata,
                     transaction_hash: _,
                     version,
-                }) => {
-                    TransactionVariant::Deploy(pathfinder_common::transaction::DeployTransaction {
+                }) if version == TransactionVersion::ZERO => TransactionVariant::DeployV0(
+                    pathfinder_common::transaction::DeployTransactionV0 {
                         contract_address,
                         contract_address_salt,
                         class_hash,
                         constructor_calldata,
-                        version,
-                    })
+                    },
+                ),
+                Transaction::Deploy(DeployTransaction {
+                    contract_address,
+                    contract_address_salt,
+                    class_hash,
+                    constructor_calldata,
+                    transaction_hash: _,
+                    version,
+                }) if version == TransactionVersion::ONE => TransactionVariant::DeployV1(
+                    pathfinder_common::transaction::DeployTransactionV1 {
+                        contract_address,
+                        contract_address_salt,
+                        class_hash,
+                        constructor_calldata,
+                    },
+                ),
+                Transaction::Deploy(DeployTransaction { version, .. }) => {
+                    panic!("unexpected deploy transaction version {version:?}")
```

Comment:
Could you add a comment stating why this is "safe"? Something along the lines of since this is technically data coming in from the gateway, panic'ing would be bad - however, since `Deploy` transactions are deprecated the only existing instances now reside on mainnet with known zero or one versions.